### PR TITLE
Move Python files to speculos/ and add setup.py

### PR DESCRIPTION
**Pull request by niooss-ledger**
This Pull Request completes the work initiated in https://github.com/LedgerHQ/speculos/pull/120 to be able to use `pip` to install Speculos. With this, using `pip install .` in the root directory of this project builds the launcher (using `cmake`) and install a `speculos` Python package, as well as a `speculos` program which lauches Speculos.

This PR:

* Modifies the `cmake` configuration to copy the built launcher and VNC server to `speculos/resources`. This enables using `cmake` with a temporary build directory which is not `build/`. Moreover this enables to package the produced binaries without embedding all the files generated by `cmake` and the C compilation process.
* Moves all Python files in a new `speculos/` directory. This is needed in order to easily install a Python package named `speculos` (otherwise `api` and `mcu` would be seen as top-level packages). Doing this, some imports were modified in order to use relative imports.
* Adds a `setup.py` (with `MANIFEST.in` to install binary files) to package Speculos.

With this, both `python3 setup.py install` and `pip install .` work, to install Speculos in a Python installation.

### CODE REVIEW

**Comment by niooss-ledger** [comment]
Also tested "pip + git repo" with a container:

```sh
podman run --rm -p 5000:5000 -ti docker.io/library/python:3.8-slim bash
apt-get update
apt-get install -y cmake gcc gcc-arm-linux-gnueabihf git libc6-dev-armhf-cross libvncserver-dev make qemu-user-static wget
pip install git+https://github.com/LedgerHQ/speculos.git@add-setup-py
wget -O btc.elf 'https://github.com/LedgerHQ/speculos/blob/master/apps/nanos%23btc%232.0%23ce796c1b.elf?raw=true'
speculos --display headless btc.elf
```

File: .github/workflows/continuous-integration-workflow.yml [hunk 1]
```diff
@@ -57,6 +57,12 @@ jobs:
         make -C build/ test
         python3 -m pytest
 
+    - name: Use pip to install speculos in a virtual environment
+      run: |
+        python3 -m venv venv-test
+        ./venv-test/bin/pip install .
```
Comment (greenknot) [inline .github/workflows/continuous-integration-workflow.yml:63]: Installing speculos on Ubuntu 20.04 thanks to pip produces an error (but doesn't make pip install fail):

```shell
$ docker run --rm -it ubuntu:20.04
root@2c48a0145b49:/# apt-get -qy update
root@2c48a0145b49:/# apt-get install -qy cmake gcc gcc-arm-linux-gnueabihf git libc6-dev-armhf-cross libvncserver-dev make qemu-user-static wget
root@2c48a0145b49:/# apt install -qy python3-pip python3-venv
root@2c48a0145b49:/# python3 -m venv venv-test
root@2c48a0145b49:/# ./venv-test/bin/pip install git+https://github.com/LedgerHQ/speculos.git@add-setup-py
...
Building wheels for collected packages: speculos, construct
  Building wheel for speculos (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /speculos/venv-test/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-3kjx96mx/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-3kjx96mx/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-jq88sw5r
       cwd: /tmp/pip-req-build-3kjx96mx/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for speculos
  Running setup.py clean for speculos
  Building wheel for construct (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /speculos/venv-test/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-k2_6x3zr/construct/setup.py'"'"'; __file__='"'"'/tmp/pip-install-k2_6x3zr/construct/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-kmz2_0b2
       cwd: /tmp/pip-install-k2_6x3zr/construct/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for construct
...
```

This patch seems to fix this error:

```patch
$ git diff
diff --git a/setup.py b/setup.py
index c0bfe8c..2cb2adf 100755
--- a/setup.py
+++ b/setup.py
@@ -53,6 +53,7 @@ setup(
         "pyqt5>=5.15.2,<6.0.0",
         "requests>=2.25.1,<3.0.0",
     ],
+    setup_requires=['wheel'],
     entry_points={
         "console_scripts": [
             "speculos = speculos.main:main",
```

**Review by greenknot — CHANGES_REQUESTED** [review]
Awesome!

File: .github/workflows/continuous-integration-workflow.yml (hunk 1 above)
Comment (niooss-ledger) [reply .github/workflows/continuous-integration-workflow.yml:63]: I was a little bit reluctant first, as using `wheel` seems to be more an acceleration (to be able to download wheels from PyPI directly) than something required, but it seems that many packages do this, cf. https://grep.app/search?q=setup_requires%3D%5B%27wheel%27%5D%2C

So I added your suggestion and tested that it worked. Thanks!

**Review by greenknot — APPROVED** [review]

**Pull request merged**